masp_primitives/src/sapling/util.rs

```diff
@@ -19,19 +19,24 @@ pub fn generate_random_rseed<P: consensus::Parameters, R: RngCore + CryptoRng>(
     height: BlockHeight,
     rng: &mut R,
 ) -> Rseed {
-    generate_random_rseed_internal(params, height, rng)
+    if params.is_nu_active(NetworkUpgrade::MASP, height) {
+        let mut buffer = [0u8; 32];
+        rng.fill_bytes(&mut buffer);
+        Rseed::AfterZip212(buffer)
+    } else {
+        Rseed::BeforeZip212(jubjub::Fr::random(rng))
+    }
 }
 
-pub(crate) fn generate_random_rseed_internal<P: consensus::Parameters, R: RngCore>(
+pub(crate) fn generate_random_rseed_internal<P: consensus::Parameters>(
     params: &P,
     height: BlockHeight,
-    rng: &mut R,
+    before: jubjub::Fr,
+    after: [u8; 32],
```

Comment:
> Both `before` and `after` are provided in BuildParams, but only one will be selected depending on the version. However, since `NetworkUpgrade::MASP` is actually activated at height 0, we don't need to do the check. Can we just use `after`? If that makes sense, we may not need both `fn output_rcm` and `fn output_rseed` in RngBuildParams either. `BeforeZip212` seems like something old that we didn't clean up, right?

Just took a quick check, and it seems the activation height depends on the network. While `NetworkUpgrade::MASP` activates on block 0 for `MainNetwork`, it seems to activate only on block 1 for `TestNetwork`. That being said, I agree with your overall point. Could we maybe remove `BeforeZip212` in a separate PR?